app/components/course-page/social-progress-modal/icon.ts

```diff
@@ -0,0 +1,43 @@
+import Component from '@glimmer/component';
+
+import discordLogoImage from 'codecrafters-frontend/images/social-icons/discord.svg';
+import linkedinLogoImage from 'codecrafters-frontend/images/social-icons/linkedin.svg';
+import slackLogoImage from 'codecrafters-frontend/images/social-icons/slack.svg';
+import twitterLogoImage from 'codecrafters-frontend/images/social-icons/twitter.svg';
+
+type Signature = {
+  Element: HTMLDivElement;
+
+  Args: {
+    onClick: (platform: string) => void;
+    platform: string;
+    selectedSocialPlatform: string;
+  };
+};
+
+export default class CoursePageSocialProgressModalIconComponent extends Component<Signature> {
+  get isSelected(): boolean {
+    return this.args.selectedSocialPlatform === this.args.platform;
+  }
+
+  get platformLogoImage(): string {
+    switch (this.args.platform) {
+      case 'discord':
+        return discordLogoImage;
+      case 'linkedin':
+        return linkedinLogoImage;
+      case 'slack':
+        return slackLogoImage;
+      case 'twitter':
+        return twitterLogoImage;
+      default:
+        return '';
```

Comment:
the default case shouldn't happen, but do you have a suggested image just in case?

Reply:
Would prevent this using types: 

```
platform: 'twitter' | 'discord' | 'linkedin' | 'slack'
```